packages/big-design/src/components/StatusMessage/spec.tsx

```diff
@@ -0,0 +1,222 @@
+import { theme } from '@bigcommerce/big-design-theme';
+import { render, screen } from '@testing-library/react';
+import React from 'react';
+import 'jest-styled-components';
+
+import { StatusMessageIcons } from './StatusMessageIcons';
+import { StatusMessagePatterns } from './StatusMessagePatterns';
+
+import { StatusMessage } from './index';
+
+test('renders basic status message', () => {
+  render(<StatusMessage message="Basic status message" />);
+
+  expect(screen.getByText('Basic status message')).toBeInTheDocument();
+});
+
+test('renders status message illustration', () => {
+  render(<StatusMessage message="Basic status message" />);
+
+  expect(screen.getByRole('figure', { hidden: true })).toBeInTheDocument();
+});
+
+test('renders panel status message', () => {
+  render(<StatusMessage heading="Panel heading" message="Basic status message" />);
+
+  expect(screen.getByRole('heading', { level: 4, name: 'Panel heading' })).toBeInTheDocument();
+});
+
+test('renders page status message', () => {
+  render(<StatusMessage heading="Page heading" message="Basic status message" size="page" />);
+
+  expect(screen.getByRole('heading', { level: 1, name: 'Page heading' })).toBeInTheDocument();
+});
+
+test('renders with button actions', () => {
+  render(
+    <StatusMessage
+      actions={[
+        { text: 'Action 1', variant: 'primary', onClick: jest.fn() },
+        { text: 'Action 2', variant: 'secondary', onClick: jest.fn() },
+      ]}
+      message="Basic status message"
+    />,
+  );
+
+  expect(screen.getByRole('button', { name: 'Action 1' })).toBeInTheDocument();
+  expect(screen.getByRole('button', { name: 'Action 2' })).toBeInTheDocument();
+});
+
+// variants
+
+// renders default variant
+test('renders with the default variant', () => {
+  render(<StatusMessage message="Basic status message" />);
+
+  const figure = screen.getByRole('figure', { hidden: true });
+
+  expect(figure).toHaveStyle({
+    backgroundImage: `radial-gradient(circle at center top, #72d8db 0%, #d9f9fa 100%)`,
+  });
+});
+
+test('renders with the 404 variant', () => {
```

Comment:
if we are just wanting to protect from accidental styling/aesthetic changes we could consider replacing these with a singular:

```tsx
const variants = [
  'info',
  'error',
  'success',
  'warning',
  'unauthorized',
  'search',
  'server-error',
  '404',
] as const;

test.each(variants)('renders with the %p variant', (variant) => {
  const { container } = render(<StatusMessage message="Basic status message" variant={variant} />);

  expect(container.firstChild).toMatchSnapshot();
});
```